specification/app/data-plane/Microsoft.App.DynamicSessions/preview/2024-10-02-preview/DynamicSessions.json

```
@@ -0,0 +1,759 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft.App Dynamic Sessions service",
+    "version": "2024-10-02-preview",
+    "x-typespec-generated": [
+      {
+        "emitter": "@azure-tools/typespec-autorest"
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{endpoint}",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "name": "endpoint",
+        "in": "path",
+        "description": "The management endpoint of the session pool.",
+        "required": true,
+        "type": "string"
+      }
+    ]
+  },
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "AadOauth2Auth": [
+        "https://dynamicsessions.io/.default"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "AadOauth2Auth": {
+      "type": "oauth2",
+      "description": "The Azure Active Directory OAuth2 Flow",
+      "flow": "accessCode",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "https://dynamicsessions.io/.default": ""
+      },
+      "tokenUrl": "https://login.microsoftonline.com/common/oauth2/token"
+    }
+  },
+  "tags": [],
+  "paths": {
+    "/execute": {
```

This is a post-action so there should be a ":" in the path before the action name. It should probably look like this:
```suggestion
    "/:execute": {
```

---

Hi, we would like to align with OpenAI apis and the usage is mostly AI related. For this particular Api, we are trying to align with https://platform.openai.com/docs/api-reference/runs/createRun, therefore there is not a ":" in the path.

---

My understanding was that this API will comply with the Azure API conventions. Has that changed?